npm/test/upload-form-files.test.ts

```diff
@@ -74,98 +127,143 @@ test("dropzone updates the page with correct number of files if only 1 file has
+  )
+
+  verifyVisibilityOfWarningMessages(mockDom.warningMessages, {
+    warningMessageElements: mockDom.warningMessages.incorrectFileExtension!,
+    expectedWarningMessageText:
+      "You can only upload a Microsoft Word file. Please try again or contact us"
+  })
+  expect(mockDom.itemRetrievalSuccessMessage).toHaveAttribute("hidden", "true")
 })
 
 test("dropzone updates the page with an error if more than 1 file has been dropped", async () => {
-  const mockDom = new MockUploadFormDom(true)
+  const mockDom = new MockUploadFormDom(
+    true,
+    1,
+    judgmentUploadPageSpecificWarningMessages()
+  )
   const dragEventClass = mockDom.addFilesToDragEvent(
-    [dummyFile, dummyFile],
+    [
+      getDummyFile("Mock File.docx", "docx"),
+      getDummyFile("Mock File.docx", "docx")
+    ],
     mockDom.dataTransferItem
   )
   const dragEvent = new dragEventClass()
   await expect(mockDom.form.handleDroppedItems(dragEvent)).rejects.toEqual(
     Error("You are only allowed to drop one file.")
   )
 
-  expect(
-    mockDom.warningMessages.incorrectItemSelectedMessage
-  ).not.toHaveAttribute("hidden", "true")
-  expect(
-    mockDom.warningMessages.submissionWithoutSelectionMessage
-  ).toHaveAttribute("hidden", "true")
+  verifyVisibilityOfWarningMessages(mockDom.warningMessages, {
+    warningMessageElements: mockDom.warningMessages.incorrectItemSelected!,
+    expectedWarningMessageText: "You can only drop a single file"
+  })
   expect(mockDom.itemRetrievalSuccessMessage).toHaveAttribute("hidden", "true")
-  expect(
-    mockDom.warningMessagesText.incorrectItemSelectedMessageText!.textContent
-  ).toStrictEqual("You can only drop a single file")
 })
 
 test("dropzone updates the page with an error if a file and a folder has been dropped", async () => {
-  const mockDom = new MockUploadFormDom(true)
+  const mockDom = new MockUploadFormDom(
+    true,
+    1,
+    judgmentUploadPageSpecificWarningMessages()
+  )
   const dragEventClass = mockDom.addFilesToDragEvent(
-    [dummyFolder, dummyFile],
+    [getDummyFolder(), getDummyFile("Mock File.docx", "docx")],
     mockDom.directoryEntry
   )
   const dragEvent = new dragEventClass()
   await expect(mockDom.form.handleDroppedItems(dragEvent)).rejects.toEqual(
     Error("You are only allowed to drop one file.")
   )
 
-  expect(
-    mockDom.warningMessages.incorrectItemSelectedMessage
-  ).not.toHaveAttribute("hidden", "true")
-  expect(
-    mockDom.warningMessages.submissionWithoutSelectionMessage
-  ).toHaveAttribute("hidden", "true")
+  verifyVisibilityOfWarningMessages(mockDom.warningMessages, {
+    warningMessageElements: mockDom.warningMessages.incorrectItemSelected!,
+    expectedWarningMessageText: "You can only drop a single file"
+  })
   expect(mockDom.itemRetrievalSuccessMessage).toHaveAttribute("hidden", "true")
+})
+
+test("dropzone updates the page with 'Only MS Word' error if 1 non-file has been dropped", async () => {
```

Comment:
Does "non-file" mean an object that is not a file, ie a folder?

Reply:
Yes. I added non-file, to emphasise that we are only accepting files, but since I'm only testing this by dragging a folder, maybe I should change it to "folder"?

Reply:
Yeah, maybe change to "folder"